### PR TITLE
Fixes #1383: Audio content highlighting

### LINKED ISSUE #1383: Audio based Content Item highlighting in StateFragment
This issue is tracker the work of Audio based content card highlighting which was previously done in https://github.com/oppia/oppia-android/pull/514 but with new ExplorationPlayer and QuestionPlayer it needs to be implemented again.

The final output should look something like this:
Screenshot_1575461301
Screenshot_1575461307

### PR DESCRIPTION
**Pull request by bhaktideshmukh**
## Explanation
Fixes #1383 

This PR covers following cases:

- Corresponding content card is highlighted, while playing audio
- Whenever audio is paused, stops highlighting
- Feedback card gets highlighted while playing audio
- Feedback card stops highlighting whenever paused
- Correct feedback starts highlighting while playing audio and stops accordingly when audio is paused
- Automatically highlight feedback item on answer selection if the audio was already playing earlier
- For multiple feedbacks only last feedback gets highlighted

**Screenshots**

||
||
||
||
||

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh I will review it tomorrow but I suggest adding a video link in description displaying all the edge cases in a single video.

**Comment by rt4914** [comment]
Pending Items / Bugs found in implementation:

1. As soon as I open the app, mainly "What is a Fraction" exploration, the content-item is highlighted by default.
2. Play the audio on state-2, now go back and notice the audio stops playing but the card is still selected.
3. When audio finishes playing and stops, the card is still highlighted.

File: app/src/main/res/layout/feedback_item.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
   <FrameLayout
     android:layout_width="wrap_content"
     android:layout_height="wrap_content"
-    android:background="@{viewModel.hasConversationView ? @drawable/content_blue_background : @drawable/transparent_background}"
+    android:background="@{viewModel.isAudioPlaying ? @drawable/content_yellow_background : @drawable/content_blue_background}"
```
Comment (rt4914) [inline app/src/main/res/layout/feedback_item.xml:20]: Changes in xml files does affect the question player too. It does mess up the entire question player/ practice mode.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

**Comment by BenHenning** [comment]
Deferring to @rt4914 to approve before I take a look (since I suspect I won't have much to add after that).

**Comment by BenHenning** [comment]
Hi. Since I was requested as a reviewer on this PR: Please note that I will be away & unavailable for reviews starting today until 23 May 2022. Please try to make any progress that you can which doesn't require me, and I'll catch up on my reviews when I return. Thanks for your flexibility--I really appreciate it.

**Pull request closed without merging**